client/src/endpoint/trades.rs

```diff
@@ -0,0 +1,211 @@
+//! Contains the endpoint for all trades.
+use error::Result;
+use std::str::FromStr;
+use stellar_resources::{AssetIdentifier, Trade};
+use super::{Body, Cursor, Direction, IntoRequest, Limit, Order, Records};
+use http::{Request, Uri};
+
+// Private struct used to define a trade pair.
+// Since the uri must include a base and a counter
+// asset, it makes sense to group them together.
+#[derive(Debug)]
+struct Tradepair {
+    base: AssetIdentifier,
+    counter: AssetIdentifier,
+}
+
+/// Represents the all trades endpoint for the stellar horizon server. The endpoint
+/// will return all trades filtered by a myriad of different query params.
+///
+/// <https://www.stellar.org/developers/horizon/reference/endpoints/trades.html>
+///
+/// ## Example
+/// ```
+/// use stellar_client::sync::Client;
+/// use stellar_client::endpoint::trades;
+///
+/// let client      = Client::horizon_test().unwrap();
+/// let endpoint    = trades::All::default();
+/// let records     = client.request(endpoint).unwrap();
+/// #
+/// # assert!(records.records().len() > 0);
+/// ```
+#[derive(Debug, Default, Cursor, Limit, Order)]
+pub struct All {
+    trade_pair: Option<Tradepair>,
+    offer_id: Option<u32>,
+    cursor: Option<String>,
+    order: Option<Direction>,
+    limit: Option<u32>,
+}
+
+impl All {
+    /// Fetches the record for a specified trade pair.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// # extern crate stellar_resources;
+    /// # extern crate stellar_client;
+    ///
+    /// use stellar_client::sync::Client;
+    /// use stellar_client::endpoint::trades;
+    /// use stellar_resources::AssetIdentifier;
+    ///
+    /// # fn main() {
+    /// let base_asset = AssetIdentifier::native();
+    /// let counter_asset = AssetIdentifier::native();
+    ///
+    /// let client      = Client::horizon_test().unwrap();
+    /// let endpoint    = trades::All::default().trade_pair(base_asset, counter_asset);
+    /// let records     = client.request(endpoint).unwrap();
+    ///
+    /// # assert_eq!(records.records().len(), 0);
```

Comment:
Here I check for a trade pair that is native, native. While this is ridiculous and obviously returns 0, I am basing this off of @kbacha's recommendation that I move away from tests that may fail in case of test net resets. **Not quite related to what you're re recommending here, just a note.**

Reply:
I thought using XLM was a cool idea that I had not thought of. But given that your tests pass and that you are asserting there is at least one trade with this pairing, it seems like people are trading xlm for xlm 🤷‍♀️ 

If you want to get deep on this you can call 

```
trades::All::default().limit(1)
```
get the asset pair off of that result, then call 
```
trades::All::default().with_trade_pair().limit(1)
``` 
using those assets and assert the returned trade has the correct asset pair.